cpp/code/basic_arrow.cc

```diff
@@ -35,6 +35,11 @@ arrow::Status ReturnNotOkMacro() {
     if (!st.ok()) {
       return st;
     }
+    auto res = builder.Finish();
+    std::shared_ptr<arrow::Array> arr;
+    if (res.ok()) {
+      arr = std::move(res).ValueUnsafe();
+    }
```

Comment:
This recipe is specifically trying to demonstrate how `Status` works and not how to construct an Array, really.

Reply:
Hmm, the section that refers this code snippet is titled "Working with Status and Result" but there's no usage of `Result` at all. That's why I added these lines to demonstrate the usage of Result.

Reply:
As much as possible I think we should try to keep each code sample focused on one thing and one thing only, hence why I'd rather not add it here. And ideally we'd have code both for this approach as well as for the macro-based approach.